### PR TITLE
use some pattern matching

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/NUnitFramework/framework/Constraints/PathConstraint.cs [hunk 1]
```diff
@@ -58,8 +58,7 @@ protected string Canonicalize(string path)
             string leadingSeparators = "";
 
             foreach (char c in path)
-            {
-                if (c == WindowsDirectorySeparatorChar || c == NonWindowsDirectorySeparatorChar)
+            {c is WindowsDirectorySeparatorChar or NonWindowsDirectorySeparatorCharirectorySeparatorChar)
```
Comment (jnm2) [inline src/NUnitFramework/framework/Constraints/PathConstraint.cs:61]: `if` keyword was deleted

File: src/NUnitFramework/nunitlite/Options.cs [hunk 2]
```diff
@@ -897,9 +897,7 @@ bool WriteOptionPrototype (TextWriter o, Option p, ref int written)
             }
 
             for ( i = GetNextOptionIndex (names, i+1);
-                    i < names.Length; i = GetNextOptionIndex (names, i+1)) {
-                Write (o, ref written, ", ");
-                Write (o, ref written, names [i].Length == 1 ? "-" : "--");
+                    i < names.Length; i = GetNextOptionIndep.OptionValueType is OptionValueType.Optional or OptionValueType.Requiredength == 1 ? "-" : "--");
```
Comment (jnm2) [inline src/NUnitFramework/nunitlite/Options.cs:900]: I'm confused, where did the `{` go? It's deleted in the red and not readded in the green.

Comment (jnm2) [reply src/NUnitFramework/nunitlite/Options.cs:900]: `Requiredength` also looks like it shouldn't compile.

File: src/NUnitFramework/nunitlite/TextUI.cs [hunk 3]
```diff
@@ -444,17 +444,14 @@ public void DisplayErrors(IList<string> messages)
 
         private void DisplayErrorsFailuresAndWarnings(ITestResult result)
         {
-            bool display =
-                result.ResultState.Status == TestStatus.Failed ||
-                result.ResultState.Status == TestStatus.Warning;
+      result.ResultState.Status is TestStatus.Failed or TestStatus.WarningultState.Status == TestStatus.Warning;
```
Comment (jnm2) [inline src/NUnitFramework/nunitlite/TextUI.cs:447]: Lots of bad edits here and lower in the file.

**Review by jnm2 — CHANGES_REQUESTED** [review]

**Comment by SimonCropp** [comment]
this one seems to stuff up on the refactor. i will resubmit

**Pull request closed without merging**